src/nebula-hooks/use-apply-column-widths.ts

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from '@nebula.js/stardust';
+import { HyperCube, ApplyColumnWidths } from '../types';
+
+export const applyColumnWidthsFactory =
+  (model: EngineAPI.IGenericObject | undefined, qHyperCube: HyperCube): ApplyColumnWidths =>
```

Comment:
Not sure I agree. The factory only takes two arguments, no grouped in an object, so I don't mind writing out the types explicitly. Maybe looks a bit cluttered since model is not always defined. And everything else is type in `ApplyColumnWIdths`. So the factory is takes two arguments and returns a function. And the line breaks makes it pretty easy to read IMO